src/query/models/tag.go

```diff
@@ -184,6 +184,18 @@ func (t Tags) IDWithExcludes(excludeKeys ...string) uint64 {
 	return h.Sum64()
 }
 
+// TagsWithoutKeys returns only the tags which do not have the given keys
+func (t Tags) TagsWithoutKeys(excludeKeys []string) Tags {
```

Comment:
Maybe just implement this and change WithoutName to call TagsWithoutKeys([]string{MetricName})

Reply:
Wouldn't that lead to weird things where whenever we call this from e.g. aggregation, we'd need to explicitly include MetricName in the list of tags to exclude?

In any case, your tag change should remove the need for this